### DISCUSSION TITLE
在 Archlinux 上使用 Magic Keyboard

### DISCUSSION THREAD

**Discussion by giscus** [Xuanwo/blog#279] [Announcements]
# 在 Archlinux 上使用 Magic Keyboard

An infrastructure engineer, focused on distributed storage system

https://xuanwo.io/2020/11-magic-keyboard-on-archlinux/

**Comment by Hanaasagi** [comment]
补充一下 "CapsLock as Ctrl" 这里，如果使用 KDE 的 keyboard 系统设置界面来配置，那么实际上修改成了 `caps:ctrl_modifier`。这个可以通过 `setxkbmap -print -verbose 10` 来查看。对于有些程式依然会存在问题的，比如 fcitx5-rime。这里需要修改成 `ctrl:nocaps` 比较好。这两个右值的作用相近，区别在于是否会有额外的 capslock 发送。可以查看定义 

- `/usr/share/X11/xkb/symbols/capslock`
```
// This changes the <CAPS> key to become a Control modifier,
// but it will still produce the Caps_Lock keysym.
hidden partial modifier_keys
xkb_symbols "ctrl_modifier" {
    replace key <CAPS> {
        type[Group1] = "ONE_LEVEL",
        symbols[Group1] = [ Caps_Lock ],
        actions[Group1] = [ SetMods(modifiers=Control) ]
    };
    modifier_map Control { <CAPS> };
};
```

- `/usr/share/X11/xkb/symbols/ctrl`
```
// Eliminate CapsLock, making it another Ctrl.
partial modifier_keys
xkb_symbols "nocaps" {
    replace key <CAPS> { [ Control_L, Control_L ] };
    modifier_map  Control { <CAPS>, <LCTL> };
};
```
